### ISSUE TITLE
New fire token not working correctly.

### ISSUE THREAD

**Issue by lifesaver666** [BartlomiejF/GaslandsBuilder#16]
Thanks for the latest changes!   I really appreciate it. But as with every new feature in just about any programs, new bugs appear.

Not sure exactly, but it seems to work fine when enabling fire, and ramping hazards up, then pressing reset, but it seems to break if reducing hazards to 0, which makes the fire turn off.   At one point, when playing around with it, the auto off stops working. and have to manually toggle it back off. It may be caused by over pressing the minus button, but I'm not sure.  There also may be a problem with cross application to other vehicles. (fire rule toggling on or off to other vehicles when fiddling with another one, but I'm not certain, might be because fire didn't toggle off correctly)

I also suggest making the black fire icon gray like the text.

**Comment by lifesaver666** [comment]
Also, I noticed the fire icon shows up on the "view" card when tapping the magnifying glass icon.

**Comment by lifesaver666** [comment]
Also, (Sorry for being so annoying, but I hope you appreciate the feedback.

It might be related to this, but I had a case where I had a problem during a game.

I had 3 vehicules, each ramped up turn by turn to gear 4, and accumulating hazards.

One of them caught fire (1st one), and I ran him over 6 hazards (up to 9) to wipe out, and reset.
So I pressed the reset button beside its hazards, and the whole sheet resetted.   All damages, gear and hazards of all vehicles went back to 0.   And oddly enough, the fire turned off on the first vehicle, but the 2nd turned on!

So all the info was wiped, but vehicle 2 caught fire somehow.

I hope you can reproduce and find what's wrong, as this app is awesome, and I have recommended it to my fellow players.  But bugs like that will prevent it from becoming more popular.   It needs to be reliable.

I hope this helps.

**Comment by BartlomiejF** [comment]
The fire token is partially fixed. Now toggling fire on one vehicle should not trigger toggling fire on other vehicles. I could not recreate the reset of whole game tracker. The resetting of fire token when hazard turns zero somehow sometimes gets stuck. Tapping the fire token (although more than once) and turning the fire off manually seems to bring the functionality back. I know that it breaks when you turn on fire, scroll down to the last vehicle and scroll back.

I am also aware that the app resets to main view when the screen is rotated so consider turning the rotation off on your device.

I'll leave the issue open as the fire token is sometimes breaking.

**Comment by lifesaver666** [comment]
Yeah. I guess what happened was that the screen rotated when I resetted a car. as I tested, rotating the screen resets the whole game.
Since in your app, rotation is useless, could rotation be disabled within the app instead of the user having to change his phone's configuration?
This would eliminate this issue, and make the app much more reliable.
